File: scripts/javascript/sync_version/time_checker.js
const request = require('sync-request');

function fetchSync(url) {
    try {
        const response = request('GET', url);
        return response.statusCode;
    } catch (error) {
        return null;
    }
}

function fetchData() {
    const url = 'https://habr.com';
    const numRequests = 100;
    const startTime = Date.now();

    for (let i = 0; i < numRequests; i++) {
        fetchSync(url);
    }

    const endTime = Date.now();
    const executionTime = (endTime - startTime) / 1000;
    console.log(`Время выполнения: ${executionTime} секунд`);
}

fetchData();

File: scripts/python/sync_version/time_checker.py
import time

import certifi
import requests


def fetch(session, url):
    response = session.get(url)
    return response.text


def main():
    url = "https://habr.com"
    num_requests = 100

    start_time = time.time()

    session = requests.Session()
    session.verify = certifi.where()

    for _ in range(num_requests):
        fetch(session, url)

    end_time = time.time()
    execution_time = end_time - start_time

    print(f"Время выполнения: {execution_time} секунд")


if __name__ == "__main__":
    main()
